src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -294,6 +324,49 @@ public void run() {
         }
     }
 
+    private void sendLocalRequest(long requestId, final String action, final TransportRequest request) {
+        final DirectResponseChannel channel = new DirectResponseChannel(action, requestId, adapter, threadPool);
+        try {
+            final TransportRequestHandler handler = adapter.handler(action);
+            if (handler == null) {
+                throw new ActionNotFoundTransportException("Action [" + action + "] not found");
+            }
+            threadPool.executor(handler.executor()).execute(new AbstractRunnable() {
+                @Override
+                protected void doRun() throws Exception {
+                    //noinspection unchecked
+                    handler.messageReceived(request, channel);
+                }
+
+                @Override
+                public boolean isForceExecution() {
+                    return handler.isForceExecution();
+                }
+
+                @Override
+                public void onFailure(Throwable e) {
+                    if (lifecycleState() == Lifecycle.State.STARTED) {
+                        // we can only send a response transport is started....
+                        try {
+                            channel.sendResponse(e);
+                        } catch (Throwable e1) {
+                            logger.warn("Failed to send error message back to client for action [" + action + "]", e1);
+                            logger.warn("Actual Exception", e);
+                        }
+                    }
+                }
+            });
+        } catch (Throwable e) {
```

Comment:
do we really need this catch here? The calling code already wraps it in Throwable catch and handles failures?